### PR TITLE
Adds Valkey support

### PR DESCRIPTION
**Pull request by jairhenrique**
_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-python-agent/blob/main/CONTRIBUTING.rst) and [code of conduct](https://github.com/newrelic/.github/blob/master/CODE_OF_CONDUCT.md)._

# Overview
Adds support to [Valkey](https://valkey.io/) an opensource fork of Redis.

# Related Github Issue
Include a link to the related GitHub issue, if applicable

# Testing
The agent includes a suite of tests which should be used to
verify your changes don't break existing functionality. These tests will run with
Github Actions when a pull request is made. More details on running the tests locally can be found
[here](https://github.com/newrelic/newrelic-python-agent/blob/main/CONTRIBUTING.rst#testing-guidelines),
For most contributions it is strongly recommended to add additional tests which
exercise your changes.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -487,6 +491,8 @@ changedir =
     datastore_pysolr: tests/datastore_pysolr
     datastore_redis: tests/datastore_redis
     datastore_rediscluster: tests/datastore_rediscluster
+    datastore_valkey: tests/datastore_valkey
+    datastore_valkeycluster: tests/datastore_valkeycluster
```
Comment (hmstepanek) [inline tox.ini:495]: It looks like this was copied over from redis but there are no valkeycluster tests in your PR.

File: tests/testing_support/db_settings.py [hunk 2]
```diff
@@ -147,6 +147,53 @@ def redis_cluster_settings():
...
+        {
+            "host": host,
+            "port": 8080 + instance_num,
+        }
+        for instance_num in range(instances)
+    ]
+    return settings
+
+
+def valkey_cluster_settings():
```
Comment (hmstepanek) [inline tests/testing_support/db_settings.py:173]: Same thing here-this isn't used anywhere.

**Review by hmstepanek — CHANGES_REQUESTED** [review]
Hey @jairhenrique thanks for adding support for this! It looks like you copy modified from redis which makes sense since Valkey is based on Redis. I need to spin up a local version of Valkey to test it out manually but I had a couple immediate things I noticed so wanted to get that feedback back to you sooner rather than later. I also made a couple commits on here to get the tests running.

File: tox.ini (hunk 1 above)
Comment (jairhenrique) [reply tox.ini]: @hmstepanek removed.

File: tests/testing_support/db_settings.py (hunk 2 above)
Comment (jairhenrique) [reply tests/testing_support/db_settings.py]: @hmstepanek removed.

**Comment by jairhenrique** [comment]
> Hey @jairhenrique thanks for adding support for this! It looks like you copy modified from redis which makes sense since Valkey is based on Redis. I need to spin up a local version of Valkey to test it out manually but I had a couple immediate things I noticed so wanted to get that feedback back to you sooner rather than later. I also made a couple commits on here to get the tests running.

That's right @hmstepanek, I made a copy of redis and removed some unnecessary checks.
The valkey project has forked the core redis libraries for nodejs and python, but they are working on an agnostic project called `valkey-glide` (rust) with bindings for those languages. I think in the future `valkey-glide` will be the core library for "all languages". It would be cool if you guys had this on your radar.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -1198,3 +1199,65 @@ jobs:
...
+
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/newrelic/newrelic-python-agent-ci:latest
+      options: >-
+        --add-host=host.docker.internal:host-gateway
+    timeout-minutes: 30
+    services:
+      valkey:
+        image: valkey/valkey
```
Comment (TimPansino) [inline .github/workflows/tests.yml:1220]: This needs a docker image tag

Comment (jairhenrique) [reply .github/workflows/tests.yml]: @TimPansino done!

**Review by hmstepanek — APPROVED** [review]
I think this looks great! I ran it locally and I see instrumentation showing up in the UI. Also ran through a diff of redis vs this and it looks good.

**Pull request merged**